edb/pgsql/compiler/relctx.py

```diff
@@ -647,23 +649,19 @@ def apply_volatility_ref(
 
 
 def ensure_transient_identity_for_path(
-        path_id: irast.PathId, stmt: pgast.BaseRelation, *,
-        ctx: context.CompilerContextLevel, type: str='int') -> None:
+    path_id: irast.PathId,
+    stmt: pgast.BaseRelation,
+    *,
+    ctx: context.CompilerContextLevel,
+) -> None:
 
-    if type == 'uuid':
-        id_expr = pgast.FuncCall(
-            name=('edgedbext', 'uuid_generate_v1mc',),
-            args=[],
-        )
-    else:
-        id_expr = pgast.FuncCall(
-            name=('row_number',),
-            args=[],
-            over=pgast.WindowDef()
-        )
+    id_expr = pgast.FuncCall(
+        name=('edgedbext', 'uuid_generate_v4',),
```

Comment:
`uuid_generate_v4` is about 2x faster than `v1`, but, being fully random, is ill-suited for persistent ids due to indexing issues.  Once we shift our Postgres baseline to 13 we can use `gen_random_uuid` instead of `uuid_generate_v4` as it's even faster.

Reply:
> Should we move to v4 more broadly?

We have to use v1mc for stored data to keep them somewhat ordered. For transient data it doesn't matter I guess, but... Why not use v1mc everywhere? Is it slower? @elprans

Reply:
(our replies crossed in time, got it re v4 being faster)